ListDetail/src/androidTest/java/com/microsoft/device/display/samples/listdetail/TopAppBarTest.kt

```diff
@@ -0,0 +1,103 @@
+import com.microsoft.device.display.samples.listdetail.ui.view.ListViewTopBar
+import com.microsoft.device.dualscreen.testutils.getString
+import com.microsoft.device.dualscreen.windowstate.WindowState
+import org.junit.Rule
+import org.junit.Test
+
+class TopAppBarTest {
+    @get: Rule
+    val composeTestRule = createAndroidComposeRule<MainActivity>()
+
+    /**
+     * Tests that app title shows in the top bar of the list view
+     */
+    @Test
+    fun listBar_showsAppTitle() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme {
+                ListViewTopBar()
+            }
+        }
+
+        composeTestRule.onNode(
+            hasText(composeTestRule.getString(R.string.app_name))
+        ).assertExists()
+    }
+
+    /**
+     * Tests that the back button is hidden in the top bar of the detail view when dual-screen
+     */
+    @Test
+    fun detailBar_buttonHiddenInDualScreenMode() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme {
+                DetailViewTopBar(isDualScreen = true)
+            }
+        }
+
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back_to_list))
+            .assertDoesNotExist()
+    }
+
+    /**
+     * Tests that the back button shows in the top bar of the detail view single screen
+     */
+    @Test
+    fun detailBar_buttonShowsInSingleScreenMode() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme {
+                DetailViewTopBar(isDualScreen = false)
+            }
+        }
+
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back_to_list))
+            .assertExists()
+    }
+
+    @Test
+    fun app_backToListButtonInSingleScreenMode() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme {
+                ListDetailApp(WindowState())
+            }
+        }
+
+        // Assert detail view is shown first
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.list_view)).assertExists()
+
+        // Click the first image from the list
+        val index = 0
+        composeTestRule.onNodeWithTag(
+            index.toString()
+        ).performClick()
+
+        // Assert the correct detail image is not shown
+        composeTestRule.onNodeWithTag(
+            composeTestRule.getString(R.string.image_tag) + index.toString()
+        ).assertExists()
+
+        // Click the back button to go back to the list view
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back_to_list)).performClick()
+
+        // Assert the detail view is now shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.detail_view)).assertDoesNotExist()
+
+        // Assert the list view is not shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.list_view)).assertExists()
+    }
```

Comment:
I think this test is missing a header comment (which I forgot to add in the DualView tests, I can go back and fix that)!
Also, some of the comments about each step have small typos - for instance, the list view is shown first (not the detail view) and some of the assertions are doing the opposite of what the comments say